File: binarySearchTree/938_range_sum_of_BST.py
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        def traverse(node):
            if not node: return 0
            
            rangeSum = 0
            if low <= node.val <= high:
                rangeSum += node.val
            
            if node.left and node.val > low:
                rangeSum += traverse(node.left)
            
            if node.right and node.val < high:
                rangeSum += traverse(node.right)
            
            return rangeSum
        
        return traverse(root)

File: binarySearchTree/938_range_sum_of_BST.go
// O(n) time | O(h) space 
// where n is number of tree nodes and h is the height of tree
func rangeSumBST(root *TreeNode, low int, high int) int {
    if root == nil {
        return 0
    }

    rangeSum := 0
    if low <= root.Val && root.Val <= high {
        rangeSum += root.Val
    }

    if root.Val > low {
        rangeSum += rangeSumBST(root.Left, low, high)
    }

    if root.Val < high {
        rangeSum += rangeSumBST(root.Right, low, high)
    }

    return rangeSum
}
